sale_order_restrict_cancel_existing_invoice/README.rst

```diff
@@ -0,0 +1,81 @@
+===========================================
+Sale Order Restrict Cancel Existing Invoice
+===========================================
+
+.. !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   !! This file is generated by oca-gen-addon-readme !!
+   !! changes will be overwritten.                   !!
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fsale--workflow-lightgray.png?logo=github
+    :target: https://github.com/OCA/sale-workflow/tree/10.0/sale_order_restrict_cancel_existing_invoice
+    :alt: OCA/sale-workflow
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/sale-workflow-10-0/sale-workflow-10-0-sale_order_restrict_cancel_existing_invoice
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/167/10.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module restrict Sale Order cancellation if a confirmed invoice linked
+to that Sale exists.
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+* Create a Sale Order and confirm it (and deliver products if needed)
+* Create Invoice and confirm it
+* On Sale Order, try to click on 'Cancel'. An error should occur
```

Comment:
Just a minor comment.

If you have a delivered picking from a SO, user wouldn't be able to cancel the this Sale Order, anyway.

Reply:
But if you haven't delivered nothing, this case applies.

Typically, that's the case of advance amount which is invoiced. Refuse to cancel such order. First, do a refund invoice (which will mark the advance invoice as paid). Then, you can cancel the sale.

Reply:
Maybe I didn't explain myself well enough. I just commented that if you follow the steps you're describing in the README file (delivering products) you'll not be able to cancel the sale order as you can not cancel it if you have delivered quantities. Anyway, this PR looks good to me 👍 . It was just a minor comment.